hazelcast/src/test/java/com/hazelcast/test/compatibility/SamplingSerializationService.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.test.compatibility;
+
+import com.hazelcast.core.ManagedContext;
+import com.hazelcast.core.PartitioningStrategy;
+import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.internal.serialization.PortableContext;
+import com.hazelcast.nio.BufferObjectDataInput;
+import com.hazelcast.nio.BufferObjectDataOutput;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.nio.serialization.PortableReader;
+import com.hazelcast.test.TestEnvironment;
+
+import java.io.IOException;
+import java.nio.ByteOrder;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static java.util.Collections.newSetFromMap;
+
+/**
+ * Serialization service that intercepts and samples serialized objects.
+ * Employed to gather samples of serialized objects used in member-to-member communication during test suite
+ * execution with mock network.
+ *
+ * @see TestEnvironment#SAMPLE_SERIALIZED_OBJECTS
+ */
+public class SamplingSerializationService implements InternalSerializationService {
+
+    static final ConcurrentMap<String, List<byte[]>> SERIALIZED_SAMPLES_PER_CLASS_NAME =
+            new ConcurrentHashMap<String, List<byte[]>>(1000);
+    // cache classes for which samples have already been captured
+    static final Set<String> SAMPLED_CLASSES = newSetFromMap(new ConcurrentHashMap<String, Boolean>(1000));
+    private static final int MAX_SERIALIZED_SAMPLES_PER_CLASS = 5;
+    // utility strings to locate test classes commonly used as user objects
+    private static final String DUMMY_CLASS_PREFIX = "Dummy";
+    private static final String TEST_CLASS_SUFFIX = "Test";
```

Comment:
These two does not look really like `SUFFIX` and `PREFIX, since you are checking them with `contains`.
Should we rename these to `INFIX` or is the check wrong ?